src/vivarium/framework/lookup/table.py

```diff
@@ -173,11 +183,19 @@ def call(self, index: pd.Index) -> pd.DataFrame:
         """
         pop = self.population_view.get(index)
         del pop["tracked"]
+        if self.clock is None:
+            raise ValueError("An interpolated table must be initialized with a clock.")
         if "year" in [col for col in self.parameter_columns]:
             current_time = self.clock()
-            fractional_year = current_time.year
-            fractional_year += current_time.timetuple().tm_yday / 365.25
-            pop["year"] = fractional_year
+            # TODO: [MIC-5478] handle Number output from clock
+            if isinstance(current_time, pd.Timestamp) or isinstance(current_time, datetime):
+                fractional_year: float = float(current_time.year)
+                fractional_year += current_time.timetuple().tm_yday / 365.25
+                pop["year"] = fractional_year
+            else:
+                raise ValueError(
+                    "The output of your clock is incompatible with your simulation data."
```

Comment:
Can you improve this error message? Maybe something like "You cannot use the column 'year' in a simulation unless your simulation uses a DateTimeClock."